### PR TITLE
👌🏻 IMPROVE: Config for memory

### PR DESCRIPTION
**Pull request by saqibameen**
Change the config of memory from: 

```js
config: {
		useGitRepo: false,
		dirToTrack: path.posix.join(''),
		extToTrack: ['*'],
	},
```

to

```ts
import { MemoryI } from '@baseai/core'

const buildMemory = (): MemoryI => ({
  // Memory metadata
  name: 'chat-with-docs',
  description: 'Chat with given docs',
  
  // Git configuration
  git: {
    enabled: true,
    include: [
      'examples/pipe.run.ts',
      '!.baseai'
    ],
    gitignore: true,
    deployedAt: 'xx',
    embeddedAt: 'xx'
  }
})

export default buildMemory
```

### CODE REVIEW

**Comment by ahmadawais** [comment]
@saqibameen let's make it simple


```js
import { MemoryI } from '@baseai/core'

const buildMemory = (): MemoryI => ({
  // Memory metadata
  name: 'chat-with-docs',
  description: 'Chat with given docs',
  
  // Git configuration
  git: {
    enabled: true,
    include: [
      'examples/pipe.run.ts',
      '!.baseai'
    ],
    gitignore: true,
    deployedAt: 'xx',
    embeddedAt: 'xx'
  }
})

export default buildMemory
```

**Comment by saqibameen** [comment]
@ahmadawais updated to match the new schema.

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
 		"clean-examples": "node .github/scripts/cleanup-examples-changesets.mjs",
 		"update-examples": "npx tsx .github/scripts/update-examples.ts",
 		"ci:version": "changeset version && node .github/scripts/cleanup-examples-changesets.mjs && pnpm install --no-frozen-lockfile",
-		"snapshot": "node .github/scripts/release-snapshot.js"
+		"snapshot": "changeset --empty && changeset version --snapshot && changeset publish --no-git-tag --snapshot"
```
Comment (ahmadbilaldev) [inline package.json:32]: I think this is the old script. Should revert these changes.

Comment (ahmadawais) [reply package.json:32]: @msaaddev what do you think? @ahmadbilaldev tag who you're asking the question from?

Comment (msaaddev) [reply package.json:32]: @ahmadbilaldev you are correct. These are old changes. Need to revert this. The following is the latest script:

```
"snapshot": "node .github/scripts/release-snapshot.js"
```

Comment (ahmadbilaldev) [reply package.json]: I have fixed it.

**Review by ahmadbilaldev — APPROVED** [review]
LGTM.
